SonarScan pipelines are failing on main

### Description

See e.g. https://github.com/moveit/moveit2/actions/runs/11868931947/job/33078707644

I'd be happy to raise a fix (green jobs make me happy), but I'm not too familiar with SonarScan, so some guidance on whether `--ignore-errors mismatch` is the fix, or whether this requires deeper investigation would be appreciated!

### ROS Distro

Humble

### OS and version

SonarScan CI workflow

### Source or binary build?

Source

### If binary, which release version?

_No response_

### If source, which branch?

rolling

### Which RMW are you using?

None

### Steps to Reproduce

Run a SonarScan job

### Expected behavior

Presumably these should not fail

### Actual behavior

They do

### Backtrace or Console output

```
Capture coverage info
  geninfo: WARNING: /usr/include/c++/13/bits/basic_string.h:181: unexecuted block on non-branch line with non-zero hit count.  Use "geninfo --rc geninfo_unexecuted_blocks=1 to set count to zero.
  geninfo: ERROR: mismatched end line for _ZN56RobotTrajectoryTestFixture_ModifyFirstWaypointByPtr_Test8TestBodyEv at /home/runner/work/moveit2/moveit2/.work/target_ws/src/moveit2/moveit_core/robot_trajectory/test/test_robot_trajectory.cpp:361: 361 -> 366
  	(use "geninfo --ignore-errors mismatch ..." to bypass this error)
  Error: Process completed with exit code 1.
```

--- a/.github/workflows/ci.yaml
+++ b/.github/workflows/ci.yaml
@@ -165,7 +165,7 @@ jobs:
         with:
           docker: $DOCKER_IMAGE
           workdir: ${{ env.BASEDIR }}/target_ws
-          lcov_capture_args: --ignore-errors=mismatch
+          lcov_capture_args: --ignore-errors=gcov,gcov,mismatch,mismatch,negative,negative
           ignore: '"*/target_ws/build/*" "*/target_ws/install/*" "*/test/*"'
       - name: Upload codecov report
         uses: codecov/codecov-action@v5

--- a/.github/workflows/ci.yaml
+++ b/.github/workflows/ci.yaml
@@ -18,7 +18,7 @@ jobs:
       matrix:
         env:
           - IMAGE: rolling-ci
-            CCOV: false # Disabled: https://github.com/moveit/moveit2/issues/2866
+            CCOV: true
             ROS_DISTRO: rolling
           - IMAGE: rolling-ci
             ROS_DISTRO: rolling

--- a/.github/workflows/ci.yaml
+++ b/.github/workflows/ci.yaml
@@ -165,6 +165,7 @@ jobs:
         with:
           docker: $DOCKER_IMAGE
           workdir: ${{ env.BASEDIR }}/target_ws
+          lcov_capture_args: --ignore-errors=mismatch
           ignore: '"*/target_ws/build/*" "*/target_ws/install/*" "*/test/*"'
       - name: Upload codecov report
         uses: codecov/codecov-action@v5

--- a/.github/workflows/sonar.yaml
+++ b/.github/workflows/sonar.yaml
@@ -103,6 +103,7 @@ jobs:
         with:
           docker: $DOCKER_IMAGE
           workdir: ${{ env.BASEDIR }}/target_ws
+          lcov_capture_args: --ignore-errors=mismatch
           ignore: '"*/target_ws/build/*" "*/target_ws/install/*" "*/test/*"'
       - name: Install ccache
         run: sudo apt install ccache build-essential

--- a/.github/workflows/sonar.yaml
+++ b/.github/workflows/sonar.yaml
@@ -103,7 +103,7 @@ jobs:
         with:
           docker: $DOCKER_IMAGE
           workdir: ${{ env.BASEDIR }}/target_ws
-          lcov_capture_args: --ignore-errors=mismatch
+          lcov_capture_args: --ignore-errors=gcov,gcov,mismatch,mismatch,negative,negative
           ignore: '"*/target_ws/build/*" "*/target_ws/install/*" "*/test/*"'
       - name: Install ccache
         run: sudo apt install ccache build-essential

